osu.Game/Overlays/Mods/ModPresetColumn.cs

```diff
@@ -44,19 +37,38 @@ protected override void LoadComplete()
         {
             base.LoadComplete();
 
-            asyncLoadPanels();
+            ruleset.BindValueChanged(_ => rulesetChanged(), true);
+        }
+
+        private IDisposable? presetSubscription;
+
+        private void rulesetChanged()
+        {
+            presetSubscription?.Dispose();
+            presetSubscription = realm.RegisterForNotifications(r =>
+                    r.All<ModPreset>()
+                     .Filter($"{nameof(ModPreset.Ruleset)}.{nameof(RulesetInfo.ShortName)} == $0"
```

Comment:
Another reason for switching to only storing `Ruleset.ShortName` would be that we wouldn't need to use the realm query language here.